drivers/usbhid-ups.c

```diff
@@ -1577,17 +1581,36 @@ static bool_t hid_ups_walk(walkmode_t mode)
 static int reconnect_ups(void)
 {
 	int ret;
+	char	*val;
+	int wait_before_reconnect = 0;
 
-	upsdebugx(4, "==================================================");
-	upsdebugx(4, "= device has been disconnected, try to reconnect =");
-	upsdebugx(4, "==================================================");
+	/* Init time to wait before trying to reconnect (seconds) */
+	val = getval(HU_VAR_WAITBEFORERECONNECT);
+	if (val) {
+		wait_before_reconnect = atoi(val);
+	}
 
+	upsdebugx(4, "Closing comm_driver previous handle");
 	/* Try to close the previous handle */
 	if (udev)
 		comm_driver->close(udev);
 
-	ret = comm_driver->open(&udev, &curDevice, subdriver_matcher, NULL);
+	if (wait_before_reconnect > 0 ) {
+		upsdebugx(4, "===================================================================");
+		upsdebugx(4, " device has been disconnected, trying to reconnect in %i seconds", wait_before_reconnect);
+		upsdebugx(4, "===================================================================");
+		sleep(wait_before_reconnect);
+		upsdebugx(4, " trying to reconnect");
+		upsdebugx(4, "===================================================================");
+	}else{
+		upsdebugx(4, "==================================================");
+		upsdebugx(4, "= device has been disconnected, try to reconnect =");
+		upsdebugx(4, "==================================================");
+	}
 
+   upsdebugx(4, "Opening comm_driver ...");
+	ret = comm_driver->open(&udev, &curDevice, subdriver_matcher, NULL);
+   upsdebugx(4, "Opening comm_driver returns ret=%i", ret);
```

Comment:
Indentation in lower two debug lines seems different from others.

Reconnection lines are formatted differently (with and without the leading `=`) and I think may be collapsed to single line of text (no `====` spacers) after the experiments.

Space near parentheses is a point for improvement :)